File: update_capabilities.js
// Script to update capability names in mockData.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'data', 'mockData.js');

// Read the file
let content = fs.readFileSync(filePath, 'utf8');

// Define capability mapping
const mappings = [
    ['CapabilityName: "Thermal Analysis"', 'CapabilityName: "Microstructure"'],
    ['CapabilityName: "Microscopy"', 'CapabilityName: "Imaging"'],
    ['CapabilityName: "Spectroscopy"', 'CapabilityName: "Small molecules"'],
    ['Capability: "Spectroscopy"', 'Capability: "Small molecules"'],
    ['CapabilityName: "Chromatography"', 'CapabilityName: "Small molecules"'],
    ['Capability: "Chromatography"', 'Capability: "Small molecules"'],
    ['CapabilityName: "Mechanical Testing"', 'CapabilityName: "Mesostructure"'],
    ['Capability: "Mechanical Testing"', 'Capability: "Mesostructure"'],
    ['CapabilityName: "Physical Testing"', 'CapabilityName: "Mesostructure"'],
    ['Capability: "Physical Testing"', 'Capability: "Mesostructure"'],
    ['Capability: "Thermal Analysis"', 'Capability: "Microstructure"'],
    ['Capability: "Microscopy"', 'Capability: "Imaging"']
];

// Apply all  replacements
mappings.forEach(([oldVal, newVal]) => {
    const regex = new RegExp(oldVal.replace(/"/g, '\\"'), 'g');
    content = content.split(oldVal).join(newVal);
});

// Write back
fs.writeFileSync(filePath, content, 'utf8');

console.log("Capability names updated successfully!");
console.log(`File: ${filePath}`);

File: update_capabilities.py
# Script to update capability names in mockData.js
import re

# Read the file
file_path = r'd:\OneDrive - The Siam Cement Public Company Limited\Desktop\PMCAL\src\data\mockData.js'

with  open(file_path, 'r', encoding='utf-8') as file:
    content = file.read()

# Define capability mapping
mappings = [
    ('CapabilityName: "Thermal Analysis"', 'CapabilityName: "Microstructure"'),
    ('CapabilityName: "Microscopy"', 'CapabilityName: "Imaging"'),
    ('CapabilityName: "Spectroscopy"', 'CapabilityName: "Small molecules"'),
    ('Capability: "Spectroscopy"', 'Capability: "Small molecules"'),
    ('CapabilityName: "Chromatography"', 'CapabilityName: "Small molecules"'),
    ('Capability: "Chromatography"', 'Capability: "Small molecules"'),
    ('CapabilityName: "Mechanical Testing"', 'CapabilityName: "Mesostructure"'),
    ('Capability: "Mechanical Testing"', 'Capability: "Mesostructure"'),
    ('CapabilityName: "Physical Testing"', 'CapabilityName: "Mesostructure"'),
    ('Capability: "Physical Testing"', 'Capability: "Mesostructure"'),
    ('Capability: "Thermal Analysis"', 'Capability: "Microstructure"'),
    ('Capability: "Microscopy"', 'Capability: "Imaging"')
]

# Apply all replacements
for old, new in mappings:
    content = content.replace(old, new)

# Write back
with open(file_path, 'w', encoding='utf-8') as file:
    file.write(content)

print("Capability names updated successfully!")
print(f"File: {file_path}")
